src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,261 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useConnect, useConnectors } from 'wagmi';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { coinbaseWalletSvg } from '../../internal/svg/coinbaseWalletSvg';
+import { defaultAvatarSVG } from '../../internal/svg/defaultAvatarSVG';
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean; // Controls the visibility state of the modal
+  onClose: () => void; // Callback function to close the modal
+  className?: string; // Optional className override for the element
+};
+
+export function WalletModal({ isOpen, onClose, className }: WalletModalProps) {
+  // We need to keep the component mounted during the exit animation
+  const [shouldRender, setShouldRender] = useState(false);
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  useEffect(() => {
+    if (isOpen) {
+      // Immediately render the modal when opening
+      setShouldRender(true);
+    } else {
+      // When closing, wait for animation to complete before removing from DOM
+      const timer = setTimeout(() => {
+        setShouldRender(false);
+      }, 100);
+      return () => clearTimeout(timer);
+    }
+  }, [isOpen]);
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl = config?.wallet?.privacyUrl ?? undefined;
+  const termsOfServiceUrl = config?.wallet?.termsUrl ?? undefined;
+
+  const handleSmartWalletConnector = useCallback(() => {
+    // WalletPreference.SMART_WALLET
+    connect({ connector: connectors[0] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleCoinbaseWalletConnector = useCallback(() => {
+    // WalletPreference.EOA
+    connect({ connector: connectors[1] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleWalletConnectConnector = useCallback(() => {
+    const walletConnectConnector = connectors.find(
+      (c) => c.name === 'WalletConnect',
+    );
+    if (walletConnectConnector) {
+      connect({ connector: walletConnectConnector });
+      onClose();
+    }
+  }, [connect, connectors, onClose]);
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      onClose();
+    }
```

Comment:
wouldn't this generally close on esc vs. enter/space?

Reply:
I was getting a lint error that required close on enter + space